### PR TITLE
mon, doc: overriding ec profile requires --yes-i-really-mean-it

### PR DESCRIPTION
**Pull request by rzarzynski**
This is per https://tracker.ceph.com/issues/64333#note-17 describing driving factors of a catastrophic cluster failure.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: doc/rados/operations/erasure-code-profile.rst [hunk 1]
```diff
@@ -96,7 +96,9 @@ Where:
 ``--force``
 
 :Description: Override an existing profile by the same name, and allow
-              setting a non-4K-aligned stripe_unit.
+              setting a non-4K-aligned stripe_unit. The former can be
+              dangerous, and thus ``--yes-i-really-mean-it`` must be
```
Comment (zdover23) [inline doc/rados/operations/erasure-code-profile.rst:100]: s/must be/must be used/
or
s/must be/must be invoked/

Comment (rzarzynski) [reply doc/rados/operations/erasure-code-profile.rst:100]: Thanks!

Comment (anthonyeleven) [reply doc/rados/operations/erasure-code-profile.rst:100]: Suggest s/The former/This/

Comment (rzarzynski) [reply doc/rados/operations/erasure-code-profile.rst:100]: The intention was call _dangerous_ just the overriding of existing, not setting an exotic stripe size.

Comment (zdover23) [reply doc/rados/operations/erasure-code-profile.rst:100]: I suggest /The former/Overriding an existing profile/

**Comment by zdover23** [comment]
jenkins test api

**Review by jdurgin — APPROVED** [review]
talking to radek, we did see a couple niche valid uses of changing a ec profile - switching from one plugin to another with the same encoding scheme (e.g. jerasure to isa for reed solomon) and possibly other parameter updates that don't affect on-disk format for other plugins. So I'm ok with keeping the command, and requiring extra confirmation.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/56434 - Squid backport
https://github.com/ceph/ceph/pull/56435 - Reef backport
Quincy backport will be done if Radek confirms that the code changes made here apply to the Quincy release branch.

**Pull request merged**